R/options_get.R

```diff
@@ -34,14 +34,24 @@ opt <- function(x, default, env = parent.frame(), ...) {
   optenv  <- get_options_env(as_env(env), inherits = TRUE)
   spec <- get_option_spec(x, env = optenv)
 
-  switch(
-    opt_source(spec, env = optenv),
+  source <- opt_source(spec, env = optenv)
+  value <- switch(
+    source,
     "envir"   = spec$envvar_fn(Sys.getenv(spec$envvar_name), spec$envvar_name),
     "option"  = getOption(spec$option_name),
     "default" = get_option_default_value(x, optenv),
     if (missing(default)) stop(sprintf("option '%s' not found.", x))
     else default
   )
+
+  spec$option_fn(
```

Comment:
I like the idea that we can pass arbitrary arguments to `option_fn` via `...`. However, if we allow for this, two other changes are required:
1. The doc `#' @param ... Additional arguments unused` must be updated in line #8;
2. A more thorough check of `option_fn` is required, otherwise the user can get weird error messages or even worse, inconsistent failures/wrong results;
3. Depending on how strict we want to be with the signature of `option_fn`, we might introduce a little overhead here and check what arguments to pass to `option_fn`:
```
.call <- as.call(
  c(
    list(spec$option_fn$fn, value),
    alist(x = x, default = default, env = env, ... = ..., source = source)[spec$option_fn$expected_args]
  )
)
eval(.call)
```

See my comment at `option_spec` for the proposed utility function.